pkg/conversion/mapi2capi/errors.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mapi2capi
+
+import (
+	"errors"
+)
+
+var (
+	errUnexpectedObjectTypeForMachine = errors.New("unexpected type for capaMachineObj")
```

Comment:
This says `capaMachineObj` though, so we should make this flexible then?

Reply:
Non blocking though, we can fix it afterwards if preferred.

Reply:
Oh, good point. fwiw, this is being used in `pkg/conversion/mapi2capi/powervs.go` (and I'm going to be using this in `pkg/conversion/mapi2capi/openstack.go` soon) so yes, we should tweak this error message.